src/og/renderer/RendererEvents.js

```diff
@@ -222,6 +224,10 @@ class RendererEvents extends Events {
                 this.mouseState.x,
                 this.mouseState.y
             );
+            this.touchState.direction = this.renderer.activeCamera.unproject(

```

Comment:
I think that this part could be optimezed. See, mouseState means that mose is always in the screen, but touch happens only with touching. So it is not necessary always making unproject, i'd better think to replace this code near touching happen triggers. What do you think?

Reply:
> 

I know.
I tried not to add touchState.direction state in handleEvents(), but there is a problem of synchronization between onTouchStart() and onTouchMove().
Next I will try to optimize this code.